src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -2425,59 +2425,71 @@ private static unsafe long ReadIdentitiesIndex(TableValueReader reader)
         {
             var items = context.Transaction.InnerTransaction.OpenTable(ItemsSchema, Items);
 
+            var results = new List<string>();
+
             var dbKey = prefix.ToLowerInvariant();
             using (Slice.From(context.Allocator, dbKey, out Slice loweredPrefix))
             {
                 foreach (var result in items.SeekByPrimaryKeyPrefix(loweredPrefix, Slices.Empty, start))
                 {
                     if (take-- <= 0)
-                        yield break;
+                        break;
 
-                    yield return GetCurrentItemKey(result.Value);
+                    results.Add(GetCurrentItemKey(result.Value));
                 }
             }
+
+            return results;
         }
 
-        public IEnumerable<string> GetDatabaseNames(TransactionOperationContext context, int start = 0, int take = int.MaxValue)
+        public List<string> GetDatabaseNames(TransactionOperationContext context, int start = 0, int take = int.MaxValue)
         {
             var items = context.Transaction.InnerTransaction.OpenTable(ItemsSchema, Items);
 
+            var names = new List<string>();
+
             const string dbKey = "db/";
             using (Slice.From(context.Allocator, dbKey, out Slice loweredPrefix))
             {
                 foreach (var result in items.SeekByPrimaryKeyPrefix(loweredPrefix, Slices.Empty, 0))
                 {
                     if (take-- <= 0)
-                        yield break;
+                        break;
 
-                    yield return GetCurrentItemKey(result.Value).Substring(3);
+                    names.Add(GetCurrentItemKey(result.Value).Substring(3));
                 }
             }
+
+            return names;
         }
 
-        public IEnumerable<DatabaseRecord> ReadAllDatabases(TransactionOperationContext context, int start = 0, int take = int.MaxValue)
+        public List<DatabaseRecord> GetAllDatabases(TransactionOperationContext context, int start = 0, int take = int.MaxValue)
         {
             var items = context.Transaction.InnerTransaction.OpenTable(ItemsSchema, Items);
 
+            var records = new List<DatabaseRecord>();
+
             const string dbKey = "db/";
             using (Slice.From(context.Allocator, dbKey, out Slice loweredPrefix))
             {
                 foreach (var result in items.SeekByPrimaryKeyPrefix(loweredPrefix, Slices.Empty, 0))
                 {
                     if (take-- <= 0)
-                        yield break;
+                        break;
 
                     var doc = Read(context, GetCurrentItemKey(result.Value));
                     if (doc == null)
                         continue;
 
-                    yield return JsonDeserializationCluster.DatabaseRecord(doc);
+                    records.Add(JsonDeserializationCluster.DatabaseRecord(doc));
```

Comment:
Can we avoid the deserialization and return the blittable?

Reply:
Likely not. You are modifying the data, after all.
And the blittable just point to the data.

Reply:
This is not used in any hot-paths at the moment. IIRC this is used in RemoveNodeFromCluster, which is not that common.